src/vm/typehashingalgorithms.h

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// ---------------------------------------------------------------------------
+// Generic functions to compute the hashcode value of types
+// ---------------------------------------------------------------------------
+
+#pragma once
+#include <stdlib.h>
+
+//
+// Returns the hashcode value of the 'src' string
+//
+inline static int ComputeNameHashCode(const SString &src)
+{
+    int hash1 = 0x6DA3B944;
+    int hash2 = 0;
+
+    for (COUNT_T i = 0; i < src.GetCount(); i += 2)
+    {
+        hash1 = (hash1 + _rotl(hash1, 5)) ^ src[i];
```

Comment:
We should base the hashcode on UTF8 strings to avoid the cost of coverting to Unicode - like what is in Fadi's checkin.